### ISSUE TITLE
supportsHighlight not working after packaging visual 

### ISSUE THREAD

**Issue by lucmax94** [Microsoft/PowerBI-visuals-tools#344]
As mentioned in the following post our visual does not work with the highlighting feature after we package the visual (pbiviz package). Everything works fine in the developer mode (pbiviz start). Is this a known issue, or are we missing something?

https://community.powerbi.com/t5/Developer/supportsHighlight-not-working-after-packaging-visual/m-p/1523090#M26625

**Comment by helen508light** [comment]
@lucmax94 
Sounds strange, could you provide the source code?

**Comment by lucmax94** [comment]
@helen508light 
I'm not able to share our source code with you, but I was able to repreduce this by creating a new visual and adding our capabilities (Not all of them). The source code and a report with some test data is attached. 

I hope this helps and thank you for looking in to it!

[Test Github.zip](https://github.com/microsoft/PowerBI-visuals-tools/files/5634097/Test.Github.zip)

**Comment by lucmax94** [comment]
@helen508light did you already have time to look at it?

**Comment by zBritva** [comment]
@lucmax94 I checked your visual, Power BI fails on creating data aggregation for the visual, I see `columnSubtotals` in capabilities, but you don't use `columns` in matrix data view. Without `columnSubtotals` it works.

**Comment by lucmax94** [comment]
@zBritva thanks for looking in to it. I tried removing `columnSubtotals`, the highlighting works fine then, but I don't get any subtotals afterwards. What I except (and need) if I remove the `columnSubtotals` is that I still get the `rowSubtotals`, but I also don't get these.

I might be missing something else, but I dont know what it is..

**Comment by zBritva** [comment]
Did you try to assign some column/bucket to columns of matrix. But without dataset. because using empty string looks strange

**Comment by lucmax94** [comment]
I've tried it and it doesn't work without adding data to the bucket. If I add data to the columns bucket everything work fine, **but** we need it to work without data in the columns bucket. I also tried a lot of other combinations of subtotals, default values in subtotals, cloumns/no columns but no one worked the way we need to.
There must be a way to only work with rows and `rowSubtotals`, have highlighting working and not include columns at all.

What I don't understand is, why it works in the development mode (pbiviz start) and breaks in the packaged visual.
